core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -567,6 +568,51 @@ default EventExecutor defaultSubscriberExecutor() {
         return FuseableStreamMessage.error(this, function);
     }
 
+    /**
+     * Peeks values emitted by this {@link StreamMessage} and applies the specified {@link Consumer}.
+     *
+     * <p>For example:<pre>{@code
+     * StreamMessage<Integer> source = StreamMessage.of(1, 2, 3, 4, 5);
+     * StreamMessage<Integer> ifEvenExistsThenThrow = source.peek(x -> {
+     *      if (x % 2 == 0) {
+     *          throw new IllegalArgumentException();
+     *      }
+     * });
+     * }</pre>
+     */
+    default StreamMessage<T> peek(Consumer<? super T> action) {
+        requireNonNull(action, "action");
+        final Function<T, T> function = obj -> {
+            action.accept(obj);
+            return obj;
+        };
+        return map(function);
+    }
+
+    /**
+     * Peeks values emitted by this {@link StreamMessage} and applies the specified {@link Consumer}.
+     *
+     * <p>For example:<pre>{@code
+     * StreamMessage<Integer> source = StreamMessage.of(1, 2, 3, 4, 5);
+     * StreamMessage<Integer> ifEvenExistsThenThrow = source.peek(x -> {
+     *      if (x % 2 == 0) {
+     *          throw new IllegalArgumentException();
+     *      }
+     * }, Integer.class);
+     * }</pre>
+     */
+    default <U extends T> StreamMessage<T> peek(Consumer<? super U> action, Class<? extends U> type) {
+        requireNonNull(action, "action");
+        requireNonNull(type, "type");
+        final Function<T, T> function = obj -> {
+            if (type.isInstance(obj)) {
+                action.accept((U) obj);
```

Comment:
Could you suppress an unsafe cast warning? It is a false positive warning.
 ```suggestion
                //noinspection unchecked
                action.accept((U) obj);
```

Reply:
I added `StreamMessage.of(0.1, 1, 0.2, 2);`!